storm-core/src/clj/backtype/storm/daemon/executor.clj

```diff
@@ -572,7 +572,8 @@
                                            (do
                                              (.put pending root-id [task-id
                                                                     message-id
-                                                                    {:stream out-stream-id :values values}
+                                                                    {:stream out-stream-id 
+                                                                     :values (if (= true (storm-conf TOPOLOGY-DEBUG)) values nil)}
```

Comment:
I would prefer to avoid calculating if TOPOLOGY-DEBUG is true each time.  This is code on the critical path so if we could do it once at the beginning of the thread function and reuse it, that would be great.